contracts/adapters/swap/drop/src/contract.rs

```diff
@@ -314,10 +314,28 @@ pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> ContractResult<Binary> {
     .map_err(From::from)
 }
 
-fn get_opposite_denom(denom: &str, remote_denom: &str, bonded_denom: &str) -> String {
-    match denom {
-        denom if denom == remote_denom => bonded_denom.to_string(),
-        denom if denom == bonded_denom => remote_denom.to_string(),
+fn get_opposite_denom_in(
+    denom_in: &str,
+    remote_denom: &str,
+    bonded_denom: &str,
+) -> ContractResult<String> {
+    match denom_in {
+        // if doing an 'in' query, only allow for simulating putting the remote denom in
+        denom_in if denom_in == remote_denom => Ok(bonded_denom.to_string()),
+        denom_in if denom_in == bonded_denom => Err(ContractError::UnsupportedDenom),
+        _ => unimplemented!(),
+    }
+}
+
+fn get_opposite_denom_out(
+    denom_out: &str,
+    remote_denom: &str,
+    bonded_denom: &str,
+) -> ContractResult<String> {
+    match denom_out {
+        // if doing an 'out' query, only allow for simulating getting the bonded denom out
+        denom_out if denom_out == remote_denom => Err(ContractError::UnsupportedDenom),
+        denom_out if denom_out == bonded_denom => Ok(remote_denom.to_string()),
```

Comment:
if they're trying to simulate a swap where where the output would be the remote denom (i.e. swapping the bonded denom to the remote denom), fail